src/lib/Group.ts

```diff
@@ -133,13 +133,19 @@ export class Group<
       async ({
         clientAddress,
         message,
+        groupId,
       }: {
         clientAddress: string
         message: DecodedMessage<ContentTypes>
+        groupId: string
       }) => {
+        // Long term these checks should be able to be done on the native layer as well, but additional checks in JS for safety
```

Comment:
Personally I think it would be better to have handling for multiple listeners/callbacks
When callbacks get registered all of the callbacks work

I think one issue is right now we can't fire "dynamic" events, they have to be predefined first

But normal RN modules I think should be fine 🤔

Reply:
Good point, we can get around the pre-registered event emitters, if we move way from expo native modules. Agree that would work as well 👌

Reply:
Hold me to it, expo removal PR by next week